tests/console/verify_efi_mok.pm

```diff
@@ -177,14 +184,14 @@ sub run {
     assert_script_run "fsck.vfat -tvV $esp_details->{partition}";
     assert_script_run "mount $esp_details->{mount}";
 
-    my $pkgs = 'efivar mokutil';
-    $pkgs .= ' dosfstools' if (is_leap('<15.2') || is_sle('<15-sp2'));
-    $pkgs .= ' pesign' unless get_var('DISABLE_SECUREBOOT', 0);
-    zypper_call "in $pkgs";
-
-    my $exp_data = get_expected_efi_settings;
+    # SUT can boot from removable (firstboot of HDD, ISO, USB bootable medium) or boot entry (non-removable)
+    # JeOS always boots firstly from removable, but the boot record will be changed to non-removable by updates
+    # Therefore the expected boot for JeOS under development and maintenance updates test slow might be different
+    # Installed SUT by YaST2 boots from non-removable by default
+    my $exp_data              = get_expected_efi_settings;
+    my $booted_from_removable = efibootmgr_current_boot()->{label} eq $exp_data->{label};
```

Comment:
I suggest to do it the other way around and consult the actual system state last, like

```
my $booted_from_removable = is_jeos;
if($booted_from_removable && is_updates_tests) {
    # Updates got installed, so it might no longer be removable
    $booted_from_removable = efibootmgr_current_boot()->{label} ne $exp_data->{label};
}

$self->verification(undef, $booted_from_removable ? undef : $exp_data, sub {
```

That way it's only run when actually used.

Also, I'm not sure whether the `eq` there is wrong - removable boot happens when there's no boot entry, so it would need to be `ne`?

Reply:
> That way it's only run when actually used.
> 
> Also, I'm not sure whether the `eq` there is wrong - removable boot happens when there's no boot entry, so it would need to be `ne`?

True, I have named the variable wrongly, should have been `non_removable`. Nice point!